### PR TITLE
Add astropy-helpers.

### PR DESCRIPTION
**Pull request by pkgw**

### CODE REVIEW

**Comment by jakirkham** [comment]
cc @astrofrog @Cadair @mwcraig

File: recipes/astropy-helpers/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed=files.txt
```
Comment (mwcraig) [inline recipes/astropy-helpers/meta.yaml]: This line should be:

```
python setup.py install --single-version-externally-managed --record=files.txt
```

File: recipes/astropy-helpers/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  commands:
+    - py.test --pyargs astropy_helpers.tests
+    - py.test --pyargs astropy_helpers.tests.test_ah_bootstrap
+    - py.test --pyargs astropy_helpers.tests.test_git_helpers
+    - py.test --pyargs astropy_helpers.tests.test_setup_helpers
+    - py.test --pyargs astropy_helpers.tests.test_utils
+
+about:
+  home: http://www.astropy.org/
+  license: BSD
```
Comment (mwcraig) [inline recipes/astropy-helpers/meta.yaml]: I believe this is `BSD 3-clause` (or you could say `Modified BSD`)

File: recipes/astropy-helpers/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  summary: Helpers for the "astropy" project.
+  dev_url: https://github.com/astropy/astropy-helpers
+
+extra:
+  recipe-maintainers:
+    - pkgw
```
Comment (mwcraig) [inline recipes/astropy-helpers/meta.yaml]: At a minimum this should also include those pinged in the comment by @jakirkham, and @bsipocz  and @pllim

**Comment by bsipocz** [comment]
While I agree with @mwcraig that it's good to move more and more packages over from the astropy channel, I wonder how large the userbase for this package actually is. 
Normally we (in astropy and in our package template) use it as a submodule, so never install it via conda or pip.

**Comment by pkgw** [comment]
I submitted this since I want to get astroquery into conda-forge, and you need it to build astroquery. My personal inclination is to have as many packages in conda-forge proper as possible, but I can imagine reasons to feel otherwise.

File: recipes/astropy-helpers/meta.yaml (hunk 1 above)
Comment (pkgw) [inline recipes/astropy-helpers/meta.yaml]: Fixed.

File: recipes/astropy-helpers/meta.yaml (hunk 2 above)
Comment (pkgw) [inline recipes/astropy-helpers/meta.yaml]: Fixed.

File: recipes/astropy-helpers/meta.yaml (hunk 3 above)
Comment (pkgw) [inline recipes/astropy-helpers/meta.yaml]: Fixed.

**Comment by bsipocz** [comment]
@pkgw - astropy-helpers should have been included in the release of astroquery as a submodule. If it isn't, then it's something we need to fix it in astroquery.

**Comment by jakirkham** [comment]
I think the test failures occur because that module is [intentionally excluded]( https://github.com/astropy/astropy-helpers/blob/v1.3.1/setup.py#L49 ) from the install. If you wish to run the tests, you could use [`test/source_files`]( https://conda.io/docs/building/meta-yaml.html#source-files ) to copy them over for the test phase.

**Comment by mwcraig** [comment]
I hadn't thought about the tests...perhaps best to simply drop those, and do a simple import test.

Regarding astroquery, and the most (or all) of the remaining astropy affiliated packages, there are two options for getting them to build, in order of preference

1. Add the `--offline` option to `python setup.py install`, which suppresses the attempt to download a newer version of `astropy_helpers`, which `conda build` will prevent (and error on). If that doesn't work...

2.  ...include `astropy_helpers` as a build, but not run, dependency. This shouldn't be necessary for any of the affiliated packages, but once upon a time there were some that did not include `astropy_helpers` as a submodule.

@pkgw, if you are interested in porting a lot of packages let me know via email. Perhaps we can coordinate the broader transition, which is long overdue...

The intent is to keep the astropy channel around, at least for now, but populate with copies of conda-forge packages.

**Comment by bsipocz** [comment]
While we always try to keep things backwards compatible, I'm not aware of
any package that tests against multiple versions of astropy-helpers. I
would advise against using any other version that was shipped with a given
release, even is it was not an official helpers release.

**Comment by astrofrog** [comment]
Here's how I've dealt with affiliated packages in the past:

https://github.com/conda-forge/wcsaxes-feedstock/blob/master/recipe/meta.yaml#L14

In my opinion, this is preferable to packaging astropy_helpers.

**Comment by pkgw** [comment]
OK, yes, it looks like we can build astroquery without needing to package this explicitly. Cool.

@mwcraig, yeah, I might be interested in helping you transition the astropy channel packages.

**Pull request closed without merging**